src/views/delete_selected_button/DeleteSelectedButton.js

```diff
@@ -1,18 +1,24 @@
-import React, { useContext } from 'react'
+import React, { useContext, useMemo } from 'react'
 import { PropTypes } from '@dhis2/prop-types'
 import i18n from '@dhis2/d2-i18n'
 import { useDataMutation } from '@dhis2/app-runtime'
 import { Button } from '@dhis2/ui'
 import { AlertContext } from '../../notifications'
 
-const mutation = {
-    resource: 'sms/inbound',
-    type: 'delete',
-    params: ({ ids }) => ({ ids }),
-}
-
-const DeleteSelectedButton = ({ selectedIds, onComplete }) => {
+const DeleteSelectedButton = ({
+    selectedIds,
+    mutationResource,
+    onComplete,
+}) => {
     const { addAlert } = useContext(AlertContext)
+    const mutation = useMemo(
+        () => ({
+            resource: mutationResource,
+            type: 'delete',
+            params: ({ ids }) => ({ ids }),
+        }),
+        [mutationResource]
+    )
```

Comment:
I would omit useMemo here until we really have a _need_ to optimize performance (and are measuring performance). Based on the recommendations here: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
You're right that this wouldn't be useful for performance reasons, I only used `useMemo` because `useDataMutation` [complains if its first arg changes between renders](https://github.com/dhis2/app-runtime/blob/15b8344b829d88552f0dae337c97f83a902585d1/services/data/src/react/hooks/useStaticInput.ts#L16).

Reply:
Ah I see. Yeah there is a canonical way of doing this with the data mutation hook where it doesn't complain. Let me see if I can find an example.

Reply:
Sorry, that took a while, I'm doing too many things at the same time. So the docs don't mention this usecase explicitly. I took a look at the source and the only parts that can be dynamic (as far as I can tell) are id, data and params. So in this case since you want to use two different endpoints, depending on the type you want to delete, I'd use two queries.

The resource part of the query/mutation being static is by design, so I think that that also fits with Austin's intented usage of this hook.